File: 조성원/[LV3] 네트워크.js
function dfs(start, n, computers, visited) {
  let visited_copy = visited.slice();

  visited_copy[start] = true;
  for (let i = 0; i < n; i++)
    if (computers[start][i] === 1 && !visited_copy[i])
      visited_copy = dfs(i, n, computers, visited_copy);

  return visited_copy;
}

function solution(n, computers) {
  let answer = 0;
  let visited = new Array(n).fill(false);

  for (let i = 0; i < n; i++) {
    if (!visited[i]) {
      visited = dfs(i, n, computers, visited);
      answer++;
    }
  }

  return answer;
}

File: 김현경/[LV3] 네트워크.py
def dfs(node, n, computers, visited):
    visited[node] = True
    for neighbor in range(n):
        if computers[node][neighbor] == 1 and visited[neighbor] == False:
            dfs(neighbor, n, computers, visited)

def solution(n, computers):
    network = 0
    visited = [False] * n
    
    for i in range(n):
        if visited[i] == False:
            dfs(i, n, computers, visited)
            network += 1
    return network